packages/amplify-cli/src/initialize-env.ts

```diff
@@ -116,6 +142,7 @@ export const initializeEnv = async (
 
     printer.success(isPulling ? '' : 'Initialized your environment successfully.');
   } catch (e) {
+    // let the error propagate up after we safely exit the spinner
     spinner.fail('There was an error initializing your environment.');
```

Comment:
we can throw Amplify Error here as well

Reply:
we are avoiding using AmplifyError unless we know resolution steps